### PR TITLE
Extend tx-related endpoints with entries resolution and filtering

### PR DESCRIPTION
**Pull request by jeroiraz**
This PR extends TxByID, VerifiedTxByID and TxScan endpoints with the capability of filtering and resolving entries.

```go
// return tx header together with resolved kv entries in a single grpc request
tx, err := client.TxByIDWithSpec(context.Background(),
			&schema.TxRequest{
				Tx: hdr.Id,
				EntriesSpec: &schema.EntriesSpec{
					KvEntriesSpec: &schema.EntryTypeSpec{
						Action: schema.EntryTypeAction_RESOLVE,
					},
				},
			})
```

### CODE REVIEW

**Comment by mmeloni** [comment]
I would add also txs batch retrivial support, with pagination.
It will be extremely useful when search cache need to start from scratch and also in hot backup.

**Comment by byo** [comment]
> I would add also txs batch retrivial support, with pagination. It will be extremely useful when search cache need to start from scratch and also in hot backup.

**Comment by jeroiraz** [comment]
> As discussed offline:
> 
> #### If no spec is present for given kind of entries, the default should be to exclude at all (both from the parsed field and the raw data in the `entries` field).
> This is necessary to avoid serious issues with backwards compatibility in the future, here's the scenario:
> 
> * immudb client uses this API call and wants to parse raw entries for KV entries but disables zset entries by using EXCLUDE set
> * if such call is made, the `entries` field will only contain raw entries for KV ones
> * in the future if we add new type of entry (like graph for example) - this means that due to lack of its spec, those entries will also be added to the `entries` list - and this would break user's assumption that only KV entries are returned
> 
> This leaves a question on how to get backward compatibility right - currently the new access method is using the same API endpoint as before meaning that with no specs at all it should return all raw entries and nothing more.

Current implementation should work as described

**Comment by jeroiraz** [comment]
> I would add also txs batch retrivial support, with pagination. It will be extremely useful when search cache need to start from scratch and also in hot backup.

sure,  stream endpoint is also useful, we can implement a new endpoint

File: pkg/api/schema/schema.proto [hunk 1]
```diff
@@ -349,24 +352,54 @@ message VerifiableZAddRequest {
...
+}
+
+message EntryTypeSpec {
+	EntryTypeAction action = 1;
+}
+
+enum EntryTypeAction {
+	EXCLUDE     = 0;
+	ONLY_DIGEST = 1;
+	RAW_VALUE   = 2;
```
Comment (byo) [inline pkg/api/schema/schema.proto:374]: how about only the referenced key info? like I want to get parsed reference but not query for the referenced value (this of course could be a future improvement)

**Review by byo — APPROVED** [review]
Did a brief check and it looks good, however I suggest a second review too.

File: pkg/api/schema/schema.proto (hunk 1 above)
Comment (jeroiraz) [reply pkg/api/schema/schema.proto:374]: yes, we can add it later on

File: pkg/database/database_test.go [hunk 2]
```diff
@@ -510,14 +511,264 @@ func TestTxByID(t *testing.T) {
...
+	t.Run("values should not be resolved but digests returned in entries field", func(t *testing.T) {
+		tx, err := db.TxByID(&schema.TxRequest{Tx: txhdr1.Id})
+		require.NoError(t, err)
+		require.NotNil(t, tx)
+		require.Len(t, tx.Entries, 1)
+		require.Empty(t, tx.KvEntries)
+		require.Empty(t, tx.ZEntries)
+
+		for _, e := range tx.Entries {
+			require.Len(t, e.Value, 0)
```
Comment (byo) [inline pkg/database/database_test.go:570]: We could also check here (and simillar checks below) if digest is not empty (or it's of the sha256 size):

```suggestion
			require.Len(t, e.Value, 0)
			require.Len(t, e.HValue, sha256.Size)
```

could even check if the sha256 is correct

File: pkg/integration/tx_entries_test.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+	"github.com/codenotary/immudb/pkg/api/schema"
+	ic "github.com/codenotary/immudb/pkg/client"
+	"github.com/codenotary/immudb/pkg/server"
+	"github.com/codenotary/immudb/pkg/server/servertest"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+func Test_GetTransactionEntries(t *testing.T) {
+	options := server.DefaultOptions()
```
Comment (byo) [inline pkg/integration/tx_entries_test.go:34]: Minor: Maybe use some temporary folder name? Using `data` sounds dangerous here.

**Review by byo — COMMENTED** [review]
One comment to resolve regarding backwards compatibility, few other minor comments.

Also make sure that we have a task to document new changes.

**Review by mmeloni — APPROVED** [review]

**Pull request merged**